### PR TITLE
Refactor minimap size to be relative to window height

### PR DESCRIPTION
**Pull request by BenoitRoux0**
Updated the minimap size to be a fraction of the window height, as opposed to a fixed size. This ensures that the minimap maintains its proportion irrespective of window size, improving user experience. The code was also made more readable by changing minimap size from a 2D vector to an integer, which simplified several calculations in minimap.c.

### CODE REVIEW

**Review by gd-harco — CHANGES_REQUESTED** [review]
Does not compile :
```
➜  cub3D git:(bonus-minimap) ✗ make bonus 
make all IS_BONUS=1
make[1] : on entre dans le répertoire « /home/gd-harco/Documents/42/level_4/cub3D »
cc -Wall -Wextra -Werror  -DWIN_WIDTH=1366 -DWIN_HEIGHT=768 -c bonus/srcs/main_bonus.c -o bonus/srcs/main_bonus.o -Ilibft/includes -Iminilibx-linux -Ibonus/incs
make[1]: ***  Aucune règle pour fabriquer la cible « bonus/srcs/minimap.o », nécessaire pour « cub3D_bonus ». Arrêt.
make[1] : on quitte le répertoire « /home/gd-harco/Documents/42/level_4/cub3D »
make: *** [Makefile:84 : bonus] Erreur 2
```

**Review by BenoitRoux0 — COMMENTED** [review]
all fixed

File: Makefile
Comment (gd-harco) [inline Makefile:1]: le makefile ici

File: Makefile [hunk 1]
```diff
@@ -36,17 +36,17 @@ endif
 
 LIBS = -lXext -lX11 -lm -lz
 
-CFLAGS =	-Wall -Wextra -Werror -g3
+CFLAGS =	-Wall -Wextra -Werror -O2 #-g3 -fsanitize=address
 
 MLX =		minilibx-linux/libmlx.a
 LIBFT =		libft/libft.a
 
 ifeq ($(shell command -v xrandr 2> /dev/null),)
```
Comment (gd-harco) [inline Makefile:44]: ici le merge a merde

**Review by gd-harco — APPROVED** [review]
Makefile ne prend pas en compte le changement de résolution de la fenêtre, je fais une PR pour fix

**Pull request merged**